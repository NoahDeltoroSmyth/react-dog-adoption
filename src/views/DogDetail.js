import React, { useEffect, useState } from 'react';
import { deleteDog, getDogsById } from '../services/dogRoute';
import { useParams, useHistory } from 'react-router-dom';
import Dogs from '../component/Dogs';
import { Link } from 'react-router-dom';

export default function DogDetail() {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const { id } = useParams();
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogsById(id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) {
    return <h1>loading</h1>;
  }

  const handleDelete = async (e) => {
    e.preventDefault();
    const { status } = await deleteDog(id);
    if (status >= 400) {
      alert('Status error');
    } else {
      alert('Successfully deleted');
      history.push('/dogs');
    }
  };

  return (
    <>
      <div className="dog-detail">
        <Dogs {...dog} showInfo={true} handleDelete={handleDelete} />
      </div>
      <div className="edit">
        <Link to={`/dogs/${id}/edit`}>Edit</Link>
      </div>
      <button className="button" onClick={handleDelete}>
        Delete
      </button>
    </>
  );
}
