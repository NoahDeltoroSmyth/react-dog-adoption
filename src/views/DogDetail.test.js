import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogDetail from './DogDetail';

it('should render a bio', async () => {
  //render
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/24']}>
      <Route path="/dogs/:id" component={DogDetail} />
    </MemoryRouter>
  );
  //wait for load
  await screen.findByText(
    'Loving and friendly, but will throw down to protect and break a finger if need be'
  );
  //take screen shot
  expect(container).toMatchSnapshot();
});
