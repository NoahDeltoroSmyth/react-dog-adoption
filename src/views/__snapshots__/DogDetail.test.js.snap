// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a bio 1`] = `
<div>
  <div
    class="dog-detail"
  >
    <div
      class="dog-card"
    >
      <h1>
        Dixie
      </h1>
      <p>
        French Bulldog
      </p>
      <img
        alt="dog-image"
        src="https://placedog.net/1000/563?id=51"
      />
      <p>
        Dixie
         is a 
        23
         year old 
        French Bulldog
      </p>
      <div>
        <p>
          Loving and friendly, but will throw down to protect and break a finger if need be
        </p>
      </div>
    </div>
  </div>
  <div
    class="edit"
  >
    <a
      href="/dogs/24/edit"
    >
      Edit
    </a>
  </div>
  <button
    class="button"
  >
    Delete
  </button>
</div>
`;
